### ISSUE TITLE
TransferActivity fails: attempts to process string as dict. 

### ISSUE THREAD

**Issue by tskluzac** [ORNL/Zambeze#170]
@renan-souza and I discovered a bug this morning on Defiant where a TransferActivity is invoked. The error we get specifically from the message_handler is 

"""
2024-05-29 11:49:57,002 - zambeze.cli_agent - INFO - [mh] Flushing activity message to flowcept
2024-05-29 11:49:57,002 - zambeze.cli_agent - ERROR - 'str' object has no attribute 'origin_agent_id'
"""

It seems that we need to alter the processing to ensure that a TransferActivity is correctly parsed. Tests are needed to ensure this doesn't happen again.

**Comment by wigging** [comment]
Can you provide a basic example that produces this error?

**Comment by tskluzac** [comment]
> Can you provide a basic example that produces this error?

Yes, just added to the description now! Long-story short -- anything that uses a standalone TransferActivity (aka the last step of any workflow where you may want to Transfer data out... does not apply to implicit data transfers that happen throughout workflow execution).

**Comment by wigging** [comment]
I'm just now digging into this. I noticed `ShellActivity` and `TransferActivity` inherit from `Activity`. Is it necessary to subclass this activity? Can the `ShellActivity` and `TransferActivity` be their own classes without having to subclass something?
